cmd/server.go

```diff
@@ -200,12 +210,13 @@ func NewServerCommand(stopCh <-chan struct{}) *cobra.Command {
 // initializeServerFlags adds the flags to <cmd>
 func initializeServerFlags(serverCmd *cobra.Command) {
 	serverCmd.Flags().IntVarP(&port, "server-port", "p", defaultServerPort, "port on which server should listen")
+	serverCmd.Flags().IntVar(&defragmentationPeriodInHour, "defragmentation-period-in-hours", 24, "period after which we should defragment etcd data directory")
```

Comment:
Should we have some validation about the full snapshot period and the defragmentation period? For example, full snapshot period should be less than the defragment period?

Reply:
> full snapshot period should be less than the defragment period

Not necessary. If there are too many `put` and `del` request in between to full snapshots, there  are huge chances of internal fragmentation leading to larger db disk size. Defragmentation in between will help in reduction of size.

Reply:
1. What happens to delta snapshots if defragmentation is done in between two full snapshots?
1. What happens to restoration in such a case?

Reply:
> What happens to delta snapshots if defragmentation is done in between two full snapshots?

Defragmentation won't loose the revision number. So, for next update events, `watch` will still send the event, delta snapshot will be taken as usual. So, object store state i.e. last full snapshot + delta snapshots over it will still be consistent.

And anyways  independent of defragmentation period smaller or larger than full snapshot period, defragmentation  step will coincide with one of the window between two consecutive full snapshot.

> What happens to restoration in such a case?

Since, restoration time there wont be any defragmentation since etcd isn''t running. Defrag request will simply fail. Sicne objects store state is stable all the time, restoration will continue as it is.

Reply:
Now I am confused. Are we talking about compaction or defragmentation?

Reply:
We are talking about defragmentation only. What is confusing you? May be you wan't to go through https://coreos.com/etcd/docs/latest/op-guide/maintenance.html once again.

Reply:
As discussed, it might be worth considering to take a full snapshot after every scheduled defragmentation (similar to what we do after a restore).

This can help avoid the edge case of restore happening between a scheduled defragmentation and the next scheduled full snapshot which can in effect nullify the defragmentation.